src/compas/geometry/_primitives/cylinder.py

```diff
@@ -0,0 +1,293 @@
+        self._height = None
+        self.plane = plane
+        self.radius = radius
+        self.height = height
+
+    @classmethod
+    def from_data(cls, data):
+        """Construct a cylinder from its data representation.
+
+        Parameters
+        ----------
+        data : :obj:`dict`
+            The data dictionary.
+
+        Returns
+        -------
+        Cylinder
+            The constructed cylinder.
+
+        Examples
+        --------
+        >>> from compas.geometry import Cylinder
+        >>> from compas.geometry import Plane
+        >>> data = {'plane': Plane.worldXY().data, 'radius': 5., 'height': 7.}
+        >>> cylinder = Cylinder.from_data(data)
+
+        """
+        cylinder = cls(Plane.worldXY(), 1, 1)
+        cylinder.data = data
+        return cylinder
+
+    # ==========================================================================
+    # descriptors
+    # ==========================================================================
+
+    @property
+    def plane(self):
+        """Plane: The plane of the cylinder."""
+        return self._plane
+
+    @plane.setter
+    def plane(self, plane):
+        self._plane = Plane(plane[0], plane[1])
+
+    @property
+    def radius(self):
+        """float: The radius of the cylinder."""
+        return self._radius
+
+    @radius.setter
+    def radius(self, radius):
+        self._radius = float(radius)
+    
+    @property
+    def height(self):
+        """float: The height of the cylinder."""
+        return self._height
+
+    @height.setter
+    def height(self, height):
+        self._height = float(height)
+
+    @property
+    def normal(self):
+        """Vector: The normal of the cylinder."""
+        return self.plane.normal
+
+    @property
+    def diameter(self):
+        """float: The diameter of the cylinder."""
+        return self.radius * 2
+
+    @property
+    def data(self):
+        """Returns the data dictionary that represents the cylinder.
+
+        Returns
+        -------
+        dict
+            The cylinder data.
+
+        """
+        return {'plane': self.plane.data,
+                'radius': self.radius}
+
+    @data.setter
+    def data(self, data):
+        self.plane = Plane.from_data(data['plane'])
+        self.radius = data['radius']
+        self.height = data['height']
+
+    def to_data(self):
+        """Returns the data dictionary that represents the cylinder.
+
+        Returns
+        -------
+        dict
+            The cylinder data.
+
+        """
+        return self.data
+
+    @property
+    def center(self):
+        """Point: The center of the cylinder."""
+        return self.plane.point
+
+    @center.setter
+    def center(self, point):
+        self.plane.point = point
+
+    @property
+    def area(self):
+        """Float: The area of the cylinder."""
+        return (pi * (self.radius**2) * 2) + (2 * pi * self.radius * self.height)
+
+    @property
+    def circumference(self):
```

Comment:
shouldn't that rather be the property of the circle

Reply:
removed